compiler/src/middle_end/matchcomp.re

```diff
@@ -485,6 +555,30 @@ let matrix_head_constructors = mtx => {
   help(mtx, []);
 };
 
+let rec pattern_head_arities_aux = (p, acc) =>
+  switch (p.pat_desc) {
+  | TPatArray(ps) when !List.mem(List.length(ps), acc) => [
+      List.length(ps),
+      ...acc,
+    ]
+  | TPatOr(p1, p2) =>
+    pattern_head_arities_aux(p1, pattern_head_arities_aux(p2, acc))
+  | _ => acc
+  };
+
+let pattern_head_arities = patt => pattern_head_arities_aux(patt, []);
```

Comment:
What's the difference between an `_aux` recursive method outside the func and the `let rec help` method like below?

Reply:
None, really. I just followed the pattern of `pattern_head_constructors`.